src/test/java/uk/gov/pay/adminusers/expungeandarchive/service/ExpungeAndArchiveHistoricalDataServiceTest.java

```diff
@@ -266,6 +292,57 @@ void shouldNotArchiveService_WhenTheLastTransactionDateIsAfterTheServicesEligibl
             expungeAndArchiveHistoricalDataService.expungeAndArchiveHistoricalData();
 
             assertFalse(serviceEntity.isArchived());
+            assertThat(serviceEntity.getFirstCheckedForArchivalDate(), is(systemDate.withZoneSameInstant(UTC)));
+            assertThat(serviceEntity.getSkipCheckingForArchivalUntilDate(), is(systemDate.plusDays(8)));
+            verifyNoMoreInteractions(mockServiceDao);
+        }
+
+        @Test
+        void shouldNotArchiveService_WhenTheFirstCheckedForArchivalDateIsAfterTheServicesEligibleForArchivingDate() {
+            when(mockExpungeAndArchiveConfig.getArchiveServicesAfterDays()).thenReturn(7);
+
+            LedgerSearchTransactionsResponse searchTransactionsResponse1 = aLedgerSearchTransactionsResponseFixture()
+                    .withTransactionList(List.of())
+                    .build();
+
+            serviceEntity = ServiceEntityFixture
+                    .aServiceEntity()
+                    .withArchived(false)
+                    .withCreatedDate(null)
+                    .withFirstCheckedForArchivalDate(systemDate)
+                    .withGatewayAccounts(List.of(gatewayAccountIdEntity1))
+                    .build();
+
+            when(mockLedgerService.searchTransactions(gatewayAccountId1, 1)).thenReturn(searchTransactionsResponse1);
+            when(mockServiceDao.findServicesToCheckForArchiving(systemDate.minusDays(7))).thenReturn(List.of(serviceEntity));
+
+            expungeAndArchiveHistoricalDataService.expungeAndArchiveHistoricalData();
+
+            assertFalse(serviceEntity.isArchived());
+            assertThat(serviceEntity.getFirstCheckedForArchivalDate(), is(systemDate.withZoneSameInstant(UTC)));
```

Comment:
Does the firstCheckedForArchivalDate get updated? Wondering why the assertion isn't 
```
assertThat(serviceEntity.getFirstCheckedForArchivalDate(), is(systemDate))
```

Reply:
`firstCheckedForArchivalDate` is only set once and won't get updated.